src/classes/OPP_AutomatedSoftCreditsService.cls

```diff
@@ -238,56 +242,29 @@ public class OPP_AutomatedSoftCreditsService {
 
 
     /*********************************************************************************************************
-    * @description Removes duplicate Relationship records based on the Contact Id.
-    * @param primaryContactToRelationships Map of Primary Contact to its Relationship records.
-    * @return Map<Id, List<npe4__Relationship__c>> Map of unique Relationship records based on Contact Id.
+    * @description Removes duplicate Relationship/Affiliation records based on the Contact Id.
+    * @param sourceIdToAssociatedRecords Map of Primary Contact/Account to its related records.
+    * @return Map<Id, List<sObject>> Map of unique Relationship records based on Contact Id.
     **********************************************************************************************************/
     @testVisible
-    private static Map<Id, List<npe4__Relationship__c>> removeDuplicateRelationships(Map<Id, List<npe4__Relationship__c>> primaryContactToRelationships) {
-        Map<Id, List<npe4__Relationship__c>> primaryContactToUniqueRelationships = new Map<Id, List<npe4__Relationship__c>>();
-
-        for (Id primaryContactId: primaryContactToRelationships.keySet()) {
-            Map<Id, npe4__Relationship__c> cleanedRelationships = new Map<Id, npe4__Relationship__c>();
-            List<npe4__Relationship__c> uniqueRelationships = new List<npe4__Relationship__c>();
-            List<npe4__Relationship__c> relationships = primaryContactToRelationships.get(primaryContactId);
-            for (npe4__Relationship__c eachRelationship : relationships) {
-                if (!cleanedRelationships.containsKey(eachRelationship.npe4__RelatedContact__c)) {
-                    cleanedRelationships.put(eachRelationship.npe4__RelatedContact__c, eachRelationship);
-                    uniqueRelationships.add(eachRelationship);
+    private static Map<Id, List<sObject>> removeDuplicateAssociatedRecords(Map<Id, List<sObject>> sourceIdToAssociatedRecords, sObjectType objectType, sObjectField fieldName) {
```

Comment:
@joy-sfdo Doesn't look like the `objectType` var is used here any more. Even the two constants for `OBJECT_TYPE_AFFILIATION` and `OBJECT_TYPE_RELATIONSHIP` aren't used either.